### PR TITLE
Fix CMake condition arg of else clause interpreted as repeated argument of opening if

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: cmake/kokkos_tribits.cmake [hunk 1]
```diff
@@ -488,7 +488,7 @@ FUNCTION(KOKKOS_LIB_COMPILE_DEFINITIONS)
   IF(KOKKOS_HAS_TRILINOS)
     #don't trust tribits to do this correctly
     KOKKOS_TARGET_COMPILE_DEFINITIONS(${TARGET} ${ARGN})
-  ELSE(TARGET ${TARGET})
+  ELSEIF(TARGET ${TARGET})
```
Comment (dalg24) [inline cmake/kokkos_tribits.cmake]: @jjwilke did you mean this or
```suggestion
  ELSE()
```
?

Comment (crtrott) [reply cmake/kokkos_tribits.cmake]: Yeah that would make more sense to me too.

**Comment by dalg24** [comment]
Actually the function is never called so I'd rather just remove it entirely.
@crtrott @jjwilke Any opinion on this?

**Comment by jjwilke** [comment]
> We can delete this - unless at some point we may want to have Kokkos -D flags. The issue is that TriBITS appends random things to pick its own names for libraries. If you call `tribits_add_library(X)` you can't just call `target_compile_definitions(X ...)`.
> 
> If we might want this logic in the future, I'd rather fix it then delete it entirely?

Addendum: I don't expect we will ever add definitions this way. We almost certainly will always put them in the config header.

**Pull request merged**